src/test/ui/borrowck/mutability-errors.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// All the possible mutability error cases.
```

Comment:
I personally am a little overwhelmed by the number of cases covered in this file.

I originally wanted to suggest that you break it up into separate files. But I can see from reading the source itself that there is a specific pattern that you are maintaining in the source, and so there is value in keeping all these cases co-located...